### ISSUE TITLE
Splitting a ready shipment creates a pending shipment

### ISSUE THREAD

**Issue by loicginoux** [solidusio/solidus#2318]
### Steps to reproduce
Via the backend:
- create an order with 1 lines item: 6 x product A
- add a customer
- create a check payment
- complete the order
- capture the payment
- go to shipment tab
- split the "ready" shipment 

### Expected behavior
The newly created shipment should be in "Ready" state

### Actual behavior
The newly created shipment is in "pending" state

### System configuration
**Solidus Version**: v2.4.0.beta1

This was not the case in v2.2.1. Can you confirm @mamhoff this is a bug ? 
Should the `FulfilmentChanger` not update the state of the new shipment depending on the order or the splitted shipment ?

**Comment by loicginoux** [comment]
The cause is actually different from my understanding: 
- in `v2.2.1`, there were this bug  #2215. The shipping method was not added to second shipment so the total of the order was not changed and the payment was kept in state "paid".
- in `v2.4.0`, this bug is fixed, so when we split the order, the new shipment creates a new shipping method and the total of the order is updated (new_order_total = old_order_total + new_shipping_methode ). The payment is passed to "balance due" so when creating the new shipment, it is not passed in state ready but stays in state "pending.

in v2.4.0, After splitting, If I pay the difference and come back tho state "paid". My 2 shipments are now ready.

Now the problem is the following, when splitting a shipment, the end client should not have to pay extra money because he will receive his order in 2 packages. Correct me if am wrong but that's actually the behaviour I am seeing with 2.4.0.

**Comment by loicginoux** [comment]
You solution would work but I don’t totally agree. From a business point of view, If the logistic department cannot ship in one package and have to send more than one package for X internal reason, the user should not have to pay for it. It's a logistic problem.
I could add an adjustment to offset the missing payment, that would work, but it should be the default behaviour. I can't imagine a case where the user will be willing to pay for that...

**Comment by tvdeyen** [comment]
> I can't imagine a case where the user will be willing to pay for that...

Amazon has this "Do you want to wait and have a single package or get items earlier, but with more packages/aka shipping costs" thing

I consider this a business problem that each store tackles differently. We should not make this decision inside of Solidus. Just add a free shipping method (backend only) an use that for the case you described.

**Comment by loicginoux** [comment]
Ok, I agree with you in the case you describe, where the user hasn't paid yet his order, and he knows that sending multiple packages will have an added cost, but in this case he chose it.
I think this is a different case when order has already been paid in full, in this case the user is not aware of the additional cost and he shouldn't because that's the problem of the seller.
Are these two cases not different ? Should not they be treated differently ?

**Comment by tvdeyen** [comment]
Nothing is true for all businesses ;)

We try to not make “right choices” in code unless necessary. It often turns out that we are wrong in doing so. 

Having free shipping for backend orders are possible. So you can easily get what you want in current versions of Solidus.

Did you try to add a backend only free shipping method? It should appear on your pending shipment as an shipping option. 

Does this work for you?

**Comment by loicginoux** [comment]
Ok I understand your point... even if I disagree :)
That's ok, no problem, we will then do it with an added adjustment or free shipping to compensate the added cost, that should work indeed.
Closing the issue.
Thank you.

**Comment by cbrunsdon** [comment]
If at all possible can you let us know exactly what you do? We really should start gathering up all the "business" problems for ben for his documentation, and if you come across a successful approach long term we can think about how better to handle/help these decisions in core.

**Comment by loicginoux** [comment]
Ok, there is some news, I think it comes from a misunderstanding from our part between shipments and cartons.

In our case, the logistic department send us a a notification with the data of what packages have been shipped/sent from the warehouse. With this data, our app updates the DB with the correct state of shipments/cartons/order.

What we did, when receiving this information, was the same as what we can do in the UI of the backend via the button "split", i.e. using the method `transfer_to_location`. That was then creating a new Shipment object, so updating the order total and we had all the problems above.

We actually changed this logic and instead created cartons for each notification received from the logistic department. This way, the shipment object stays unique, the order total is not updated and the price stays the same. We now do that via `Spree::OrderShipping.new(order).ship(...)`.

So the problem is that we didn't understand what the "Split" button really meant from the business perspective. Actually, the "shipments" tab in backend order page has to be seen as a way to visualize and create "sub-order" via the split action, in the case when there is more than one wharehouse (or expedition point). 

Seing it this way the solidus behaviour is correct. Hopefully, with the new documentation, there will be less misunderstanding like these. Thanks for yout time 👍

**Comment by tvdeyen** [comment]
Thanks for the clarification. Yes, this is a confusing part of the system. 

Actually this reminds me that I wanted to continue working on changing the shipments order view so it does less confusing things.

Great that you could make it working for you. 👍

**Comment by sebfie** [comment]
I implemented what @loicginoux said and in fact we have for the moment one issue : 

When sending a carton with partial products, the shipment switch to state "pending" instead of stay in state "ready".

I think it's due to this code (perhaps): 

https://github.com/solidusio/solidus/blob/v2.2/core/app/models/spree/shipment.rb#L81

It tooks all `inventory_units.all?` instead of `inventory_units. on_hand?` maybe? 

I did not have time to debug this part, but I will do it shortly, and maybe make a PR for that.

Thx for your time !
